File: BackEnd/Mmo/Mmo_Domain/ModelResponse/OrderAdminResponse.cs
namespace Mmo_Domain.ModelResponse;

public class OrderAdminResponse
{
    public int OrderId { get; set; }
    public string ShopName { get; set; } = null!;
    public decimal TotalPrice { get; set; }
    public int Quantity { get; set; }
    public string BuyerName { get; set; } = null!;
    public string SellerName { get; set; } = null!;
    public string? Status { get; set; }
    public DateTime? OrderDate { get; set; }
}

File: frontend/src/models/modelResponse/OrderAdminResponse.ts
export interface OrderAdminResponse {
  orderId: number;
  shopName: string;
  totalPrice: number;
  quantity: number;
  buyerName: string;
  sellerName: string;
  status: string;
  orderDate: string;
  productName?: string;
  productVariantName?: string;
  productVariantId?: number;
}

export interface AccountStorageInfo {
  storageId: number;
  username: string;
  password: string;
  status: boolean;
}

export interface OrderDetailResponse {
  orderId: number;
  productName?: string;
  productVariantName?: string;
  quantity: number;
  totalPrice: number;
  status?: string;
  orderDate?: string;
  payload?: string;
  accounts?: AccountStorageInfo[];
}
